source/common/formatter/substitution_formatter.cc

```diff
@@ -1918,6 +1925,69 @@ UpstreamHostMetadataFormatter::UpstreamHostMetadataFormatter(const std::string&
                           return host->metadata().get();
                         }) {}
 
+ProxyProtocolTlvsFormatter::ProxyProtocolTlvsFormatter(const std::string& tlv_type) {
+  ASSERT(!tlv_type.empty());
```

Comment:
This `ASSERT` guarantees that the string is not empty, but yes, it should be guaranteed upon envoy startup when we read the config. For example, if no param is provided (i.e. `"%PROXY_PROTOCOL_TLVS()%"`), we fail as follows: 
```
[2023-01-23 20:09:49.573][29860][critical][main] [source/server/server.cc:117] error initializing configuration '/home/ubuntu/serverless-proxy/examples/proxy_protocol_client_server_poc/envoy/hcm_bootstrap.yaml': PROXY_PROTOCOL_TLVS requires parameters
```

I added the `ASSERT` here to copy how it's used [here](https://github.com/10gen/envoy-serverless/blob/aa484fbfcfd6aa19544a860d45e9794ff0f2ee73/source/common/formatter/substitution_formatter.cc#L2142) to validate the same type of variable. The `EnvironmentFormatter` also requires a param, so it would also fail upon envoy startup without one provided. Maybe it's included there to be extra safe because `.c_str()` is called on the variable [here](https://github.com/10gen/envoy-serverless/blob/aa484fbfcfd6aa19544a860d45e9794ff0f2ee73/source/common/formatter/substitution_formatter.cc#L2144), but I'm not sure why it's necessary there either 🤷‍♀️  

Happy to remove it if you think it's redundant!

Reply:
No. This looks great.